js/ui/appActivation.js

```diff
@@ -90,6 +90,48 @@ const AppActivationContext = new Lang.Class({
             this.showSplash();
     },
 
+    // Activate an app via a D-Bus call and keep track of its spalsh screen
+    // if we had to open a new window for it. Otherwise, if there were
+    // windows already open, just do the D-Bus call
+    activateViaDBusCall: function(busName, path, interfaceName, method, args, done) {
+        Gio.bus_get(Gio.BusType.SESSION, null, Lang.bind(this, function(source, result) {
+            let bus = null;
+            try {
+                bus = Gio.bus_get_finish(result);
+            } catch (error) {
+                done(error, null);
+                return;
+            }
+
+            let proxy = new Gio.DBusProxy({
+                g_bus_type: Gio.BusType.SESSION,
+                g_connection: bus,
```

Comment:
Damn, I had a try of this and it doesn't seem to work :( I've left the old code in place for now.